docs/interop.md

```diff
@@ -108,8 +114,21 @@ internal unsafe interface IServiceProviderInterop
     void GetRuntimeClassName(out IntPtr className);
     void GetTrustLevel(out TrustLevel trustLevel);
 
+    // Note: methods should be typed 'void' with the CLR mapping exceptions to HRESULTs.
+    // Parameter and return values should use ABI types, versus projected types,
+    // and these should be manually marshaled.
+    // BEFORE: object GetService(Type serviceType);
     void GetService(IntPtr type, out IntPtr service);
 }
+
+// ...
+
+// Cast a class or object to an interop interface using the .As<>() method described above
+var serviceProvider = someObject.As<IServiceProviderInterop>();
+IntPtr service;
+serviceProvider.GetService(IntPtr.Zero, out service);
+// See Create RCW section below for marshaling an object pointer from the ABI 
+
 ```
 
 ## Create RCW

```

Comment:
Do we want to add details on when the respective FromAbi function should be used (i.e. ABI.Namespace.T.FromABI vs MarshalGeneric<T>.FromABI).   The other 2 are pretty obvious.